flexmeasures/ui/tests/test_user_crud.py

```diff
@@ -1,109 +1,81 @@
-from flask import url_for, request
+from flask import url_for
+import pytest
+
 from flexmeasures.data.services.users import find_user_by_email
+from flexmeasures.ui.tests.utils import mock_user_response
+
+"""
+Testing if the UI crud views do auth checks and display answers.
+Actual logic is tested in the API tests.
+"""
 
 
-def test_user_crud_as_non_admin(client, as_prosumer):
-    user_index = client.get(url_for("UserCrud:index"), follow_redirects=True)
+@pytest.mark.parametrize(
+    "view", ["index", "get", "toggle_active", "reset_password_for"]
+)
+def test_user_crud_as_non_admin(client, as_prosumer, view):
+    user_index = client.get(url_for("UserCrudUI:index"), follow_redirects=True)
     assert user_index.status_code == 403
     prosumer2_id = find_user_by_email("test_prosumer2@seita.nl").id
     user_page = client.get(
-        url_for("UserCrud:get", id=prosumer2_id), follow_redirects=True
-    )
-    assert user_page.status_code == 403
-    user_page = client.get(
-        url_for("UserCrud:toggle_active", id=prosumer2_id), follow_redirects=True
-    )
-    assert user_page.status_code == 403
-    user_page = client.get(
-        url_for("UserCrud:delete_with_data", id=prosumer2_id), follow_redirects=True
-    )
-    assert user_page.status_code == 403
-    user_page = client.get(
-        url_for("UserCrud:reset_password_for", id=prosumer2_id), follow_redirects=True
+        url_for(f"UserCrudUI:{view}", id=prosumer2_id), follow_redirects=True
     )
     assert user_page.status_code == 403
 
 
-def test_user_list(client, as_admin):
-    user_index = client.get(url_for("UserCrud:index"), follow_redirects=True)
+def test_user_list(client, as_admin, requests_mock):
+    requests_mock.get(
+        "http://localhost//api/v2_0/users",
+        status_code=200,
+        json=mock_user_response(multiple=True),
+    )
+    user_index = client.get(url_for("UserCrudUI:index"), follow_redirects=True)
     assert user_index.status_code == 200
     assert b"All active users" in user_index.data
-    assert b"test_prosumer@seita.nl" in user_index.data
-    assert b"test_prosumer2@seita.nl" in user_index.data
+    assert b"alex@seita.nl" in user_index.data
+    assert b"bert@seita.nl" in user_index.data
 
 
-def test_user_page(client, as_admin):
-    prosumer2 = find_user_by_email("test_prosumer2@seita.nl", keep_in_session=False)
-    user_page = client.get(
-        url_for("UserCrud:get", id=prosumer2.id), follow_redirects=True
+def test_user_page(client, as_admin, requests_mock):
+    mock_user = mock_user_response(as_list=False)
+    requests_mock.get(
+        "http://localhost//api/v2_0/user/2", status_code=200, json=mock_user
```

Comment:
Will these only test v2_0?

Reply:
Yes, but it's the same issue you brought up in another comment.